torch/prototypes/nestedtensor/__init__.py

```diff
@@ -0,0 +1,64 @@
+import torch
+from . import nested
+import torch.prototypes.nestedtensor.codegen as codegen
+
+def is_available():
+    return hasattr(torch._C, "_nestedtensor_init")
+
+if is_available() and not torch._C._nestedtensor_init():
+    raise RuntimeError("Failed to initialize PyTorch nestedtensor support")
+
+import torch.prototypes.nestedtensor as nestedtensor
+
+NestedTensor = nested.NestedTensor
+
+def _create_out(input1, out, dtype=None):
+    if out is None:
+        if dtype is None:
+            dtype = input1.dtype
+        out_tensors = []
+        for tensor in input1.tensors:
+            out_tensors.append(torch.empty_like(tensor, dtype=dtype))
+        out = NestedTensor(out_tensors)
+    assert len(out) == len(input1)
+    return out
```

Comment:
Should this error out if the dtype is different?

Reply:
No, because dtype might be set to something that doesn't match input1's dtype.